synapse/storage/util/id_generators.py

```diff
@@ -482,3 +446,58 @@ def _add_persisted_position(self, new_id: int):
                 # There was a gap in seen positions, so there is nothing more to
                 # do.
                 break
+
+
+@attr.s(slots=True)
+class _AsyncCtxManagerWrapper:
+    """Helper class to convert a plain context manager to an async one.
+    """
```

Comment:
So this wrapper takes something that isn't async and exposes it as if it is? The reason we do that is to make the interfaces consistent?

Reply:
Yup, eactly